rdr_service/dao/study_nph_sms_dao.py

```diff
@@ -229,45 +229,21 @@ def source_data(self, **kwargs):
             if 'pbrc' in kwargs.get('recipient').lower():
                 query = query.add_columns(
                     SmsSample.body_weight_kg,
-                    case(
-                        [(
-                            func.json_length(aliquot_sample.supplemental_fields) > 0,
-                            func.json_extract(
-                                aliquot_sample.supplemental_fields, '$.dlwDose.batchid'
-                            )
-                        )],
-                        else_=func.json_extract(
-                            parent_fields.supplemental_fields, '$.dlwDose.batchid'
-                        )
-                    ).label('dlw_dose_batch'),
-                    case(
-                        [(
-                            func.json_length(aliquot_sample.supplemental_fields) > 0,
-                            func.json_extract(
-                                aliquot_sample.supplemental_fields, '$.dlwDose.doseAdministered'
-                            )
-                        )],
-                        else_=func.json_extract(
-                            parent_fields.supplemental_fields, '$.dlwDose.doseAdministered'
-                        )
-                    ).label('dlw_dose_date_time'),
-                    case(
-                        [(
-                            func.json_length(aliquot_sample.supplemental_fields) > 0,
-                            func.json_extract(
-                                aliquot_sample.supplemental_fields, '$.dlwDose.dose'
-                            )
-                        )],
-                        else_=func.json_extract(
-                            parent_fields.supplemental_fields, '$.dlwDose.dose'
-                        )
-                    ).label('dlw_dose_grams')
+                    DlwDosage.batch_id.label('dlw_dose_batch'),
+                    DlwDosage.dose_time.label('dlw_dose_date_time'),
+                    DlwDosage.calculated_dose.label('dlw_dose_grams')
                 ).outerjoin(
                     aliquot_sample,
                     SmsSample.sample_id == aliquot_sample.aliquot_id
                 ).outerjoin(
                     parent_fields,
                     aliquot_sample.parent_sample_id == parent_fields.id
+                ).outerjoin(
+                    Participant,
+                    cast(Participant.biobank_id, String) == func.substr(SmsN0.biobank_id, 2)
+                ).outerjoin(
+                    DlwDosage,
+                    and_(Participant.id == DlwDosage.participant_id, DlwDosage.ignore_flag == 0)
```

Comment:
I think we need a condition for `DlwDosage.visit_period` here as well, since there can be multiple DlwDosages per participant ID. That value should match the visit period on the sample's `Order`. The issue is that the names  HPRO sends us in the DLW API isn't the same string value that HPRO sends us in the Order API. You'll probably need to map your `VisitPeriod` enum for the 3 Diets to the 3 visit period strings.
Let me know if you want to chat about this for more details.